nncf/common/pruning/utils.py

```diff
@@ -187,6 +192,93 @@ def get_previous_conv(graph: NNCFGraph, nncf_node: NNCFNode,
+            name = node.ia_op_exec_context.scope_in_model if hasattr(node, 'ia_op_exec_context') \
+                else get_original_node_name(node.node_name)
+            if name in in_channels and name in out_channels:
+                continue
+            in_channels[name] = node.module_attributes.in_channels
+            out_channels[name] = node.module_attributes.out_channels
+    return in_channels, out_channels
+
+
+def get_cluster_next_nodes(graph: NNCFGraph, pruned_groups_info,
+                           prunable_types: List[str]) -> Dict[int, List[str]]:
+    next_nodes = {}
+    for cluster in pruned_groups_info.get_all_clusters():
+        next_nodes_cluster = set()
+        cluster_nodes = set()
+        for cluster_node in cluster.nodes:
+            nncf_cluster_node = graph.get_node_by_id(cluster_node.nncf_node_id)
+            nncf_cluster_node_scope = nncf_cluster_node.ia_op_exec_context.scope_in_model \
+                if hasattr(nncf_cluster_node, 'ia_op_exec_context') \
+                else get_original_node_name(nncf_cluster_node.node_name)
+            cluster_nodes.add(nncf_cluster_node_scope)
+            curr_next_nodes = get_next_nodes_of_types(graph, nncf_cluster_node, prunable_types)
+
+            next_nodes_idxs = [n.ia_op_exec_context.scope_in_model if hasattr(n, 'ia_op_exec_context')
+                               else get_original_node_name(n.node_name) for n in curr_next_nodes]
+            next_nodes_cluster = next_nodes_cluster.union(next_nodes_idxs)
+        next_nodes[cluster.id] = list(next_nodes_cluster - cluster_nodes)
+    return next_nodes
+
+
+def count_flops(graph: NNCFGraph, input_shapes: dict, output_shapes: dict,
+                conv_op_types: List[str], linear_op_types: List[str],
+                input_channels: dict = None, output_channels: dict = None):
+    flops = {}
+    input_channels = input_channels or {}
+    output_channels = output_channels or {}
+    for node in graph.get_nodes_by_types(conv_op_types):
+        name = node.ia_op_exec_context.scope_in_model if hasattr(node, 'ia_op_exec_context') \
+            else get_original_node_name(node.node_name)
+        if name in flops:
+            continue
+        num_in_channels = input_channels.get(name, node.module_attributes.in_channels)
+        num_out_channels = output_channels.get(name, node.module_attributes.out_channels)
+        flops[name] = 2 * np.prod(node.module_attributes.kernel_size) * \
+                      num_in_channels * num_out_channels * np.prod(output_shapes[name])
+
+    for node in graph.get_nodes_by_types(linear_op_types):
+        name = node.ia_op_exec_context.scope_in_model if hasattr(node, 'ia_op_exec_context') \
+            else get_original_node_name(node.node_name)
+        flops[name] = 2 * np.prod(input_shapes[name]) * np.prod(output_shapes[name])
+
+    return flops
+
+
+def calculate_in_out_channel_in_uniformly_pruned_model(pruning_groups, pruning_rate,
```

Comment:
`in_out_channels`?

Reply:
@mkaglins Just a shortening for input/output, since the name of the function is already quit long. Do you have any suggestions how to rename it?

Reply:
It was a suggestion to rename `in_out_channel` -> `in_out_channels` here.